doc/sel_temperature_sensor.rst

```diff
@@ -70,40 +90,41 @@ The temperature value is nine characters long and in the fixed form:
     - 'n' is '0' thru '9'.
     - '.' decimal point character is fixed in the fifth position of the value.
 
-Channel Disconnected Value
---------------------------
-The instrument can report an specific value for any discoinnected temperature channel.
-Disconnected temperature channel value: '999.9990'
-
-Channel Error Value
--------------------
-The instrument can report an error for any temperature channel.
-Temperature channel error value: '-201.0000'
+Channel Error or Disconnected Value
+-----------------------------------
+The instrument can report an error for any temperature channel. The error value commonly represents that the sensor is disconnected.
+Due to the nature of operation of thermocouple and RTD measurements, the instrument error value differs between the two instrument types.
+A thermocouple instrument reports an error value of: '-201.0000'
+A RTD instrument reports an error value of: '9999.9990'
```

Comment:
Do you map both of these to the DISCONNECTED_VALUE that you defined in ts_ess_common, or to `nan`?

I doubt it because I don't see any use of -201 in this code.

By having the server report different invalid values for different sensors, you are making life very difficult for the client. I strongly urge you to map sensor-specific invalid values to some common invalid value: `nan` would be the obvious choice.

With such a change I think you could get rid of DISCONNECTED_VALUE from lsst.ts.ess.common and have a set of device-specific values in this package.

Reply:
It also suggests you should have two different classes for thermocouple vs RTD temperature sensors, for two reasons: the different invalid value and the different # of channels.

Or at least one class with a constructor argument set by sensor configuration, so you can specify which is which.

The doc string for TemperatureSensor says: "Perform protocol conversion for RTD thermal sensors" which implies that it doesn't work for thermocouple sensors. So maybe you already have a 2nd class that I missed?

Reply:
Trivial correction: `A RTD` -> `An RTD`

Reply:
There was confusion about the values reported which has been cleared up by now. Garry has been preparing documentation improvements and I prefer to wait until then to clear up this part of the code.

Having said that, the temperature sensor code does check for the DISCONNECTED_VALUE (of 9999.9990) and replaces that with math.nan. I do like your suggestion to remove the constant from ts_ess_common and to have that as a local constant in the temperature sensor class. Whether or not two classes are needed is something I'd like to think about.

I made the trivial correction.